lightning/src/ln/channel_id.rs

```diff
@@ -0,0 +1,144 @@
+// This file is Copyright its original authors, visible in version control
+// history.
+//
+// This file is licensed under the Apache License, Version 2.0 <LICENSE-APACHE
+// or http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your option.
+// You may not use this file except in accordance with one or both of these
+// licenses.
+
+//! ChannelId definition.
+
+use crate::ln::msgs::DecodeError;
+use crate::sign::EntropySource;
+use crate::util::ser::{Readable, Writeable, Writer};
+
+use bitcoin::hashes::hex::ToHex;
+
+use crate::io;
+use crate::prelude::*;
+use core::fmt;
+use core::ops::Deref;
+
+/// A unique 32-byte identifier for a channel.
+/// Depending on how the ID is generated, several varieties are distinguished (but all are stored as 32 bytes):
+/// - v1: generated based on funding tx outpoint (txid&index)
+/// - temporary: generated randomly
+/// (later planned v2: based on revocation point)
+/// The variety (context) is not stored, it is relevant only at creation.
+/// This is not exported to bindings users as we just use [u8; 32] directly
+#[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct ChannelId {
```

Comment:
Valid point! It would simplify the change.
I follow the opinion of the maintainers, as you have a much better feel of the overall style/approach of the whole project than me (and also of Rust probably).
My driver was that I find `id.0` less readable/typesafe than `id.bytes()` -- but this is subjective, and that it is easier to add fields once creation is behind constructors -- but there's little chance foreseen and it can be changed later if needed.
I think I'll rework as suggested.

Reply:
`.bytes()` has not much value in that case; I would keep the `from_bytes()` generic ctor, to suggest that constructor should be used, and have a place to insert the comment to use the other specific ctors whenever possible :)